### ISSUE TITLE
Replace file uploads with modern upload method (XMLHttpRequest Level 2)

### ISSUE THREAD

**Issue by felicitus** [partkeepr/PartKeepr#407]
See https://github.com/ivan-novakov/extjs-upload-widget

Of course, we need to check for supported browsers.

**Comment by DIYsciBorg** [comment]
Please remember to keep the url field too though! generally, I am already looking at the distributor web site when I am entering part data, and its WAY easier to just copy the image url than to download and then upload to partkeepr.

**Comment by felicitus** [comment]
@DIYsciBorg absolutely! I'm using that feature often as well, so no worries, this will stay ;)
